support/types.js

```diff
@@ -238,10 +238,21 @@ exports.isDataView = isDataView;
 function isSharedArrayBufferToString(value) {
   return ObjectToString(value) === '[object SharedArrayBuffer]';
 }
-isSharedArrayBufferToString.working = (
-  typeof SharedArrayBuffer !== 'undefined' &&
-  isSharedArrayBufferToString(new SharedArrayBuffer())
-);
+// Avoid invoking SharedArrayBuffer constructor until required, then memoize
+Object.defineProperty(isSharedArrayBufferToString, 'working', {
+  get: (function() {
+    var isWorking;
+    return function () {
+      if (isWorking === undefined) {
+        isWorking = (
+          typeof SharedArrayBuffer !== 'undefined' &&
+          isSharedArrayBufferToString(new SharedArrayBuffer())
```

Comment:
the problem with this approach is that later-run code can `delete SharedArrayBuffer` and cause this check to fail, incorrectly.

in other words, the check *must* be done eagerly, on first load. It's totally fine, however, to cache `SharedArrayBuffer` somehow at module level, and only do the constructor check when the function is invoked, if that would help fix the false positive.

Reply:
@ljharb I see what you're saying with regard to deleting the global `SharedArrayBuffer`. Any buffers created between the time we initialize `util` and when we delete `SharedArrayBuffer` would be reported incorrectly.

Would that same later-run code deleting `SharedArrayBuffer` not also affect the current implementation because of these lines here though?

https://github.com/browserify/node-util/blob/4b1c0c79790d9968eabecd2e9c786454713e200f/support/types.js#L245-L248

```js
var util = require('./node_modules/util');
var buffer = new SharedArrayBuffer();
delete SharedArrayBuffer;
util.types.isSharedArrayBuffer(buffer)
// false
```

We can fix both that issue and the one that I was going to introduce by doing as you suggested and storing a copy of `SharedArrayBuffer` (as the oh-so-creatively-named `SharedArrayBufferCopy`) which fixes the case mentioned above.